Makefile

```diff
@@ -28,66 +28,62 @@ endif
 # Some will have dedicated targets to make it easier to type, for example
 # "apiserver" instead of "bin/apiserver".
 #########################################################################
-build: init \
+build: .init \
        $(BINDIR)/controller $(BINDIR)/registry $(BINDIR)/k8s-broker \
        $(BINDIR)/service-catalog $(BINDIR)/user-broker \
        $(BINDIR)/apiserver
 
 controller: $(BINDIR)/controller
-$(BINDIR)/controller: pkg/controller/catalog
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/controller: pkg/controller/catalog $(shell find pkg/controller/catalog -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/pkg/controller/catalog
 
 registry: $(BINDIR)/registry
-$(BINDIR)/registry: contrib/registry
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/registry: contrib/registry $(shell find contrib/registry -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/contrib/registry
 
-$(BINDIR)/k8s-broker: contrib/broker/k8s
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/k8s-broker: contrib/broker/k8s $(shell find contrib/broker/k8s -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/contrib/broker/k8s
 
-$(BINDIR)/service-catalog: cmd/service-catalog
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/service-catalog: cmd/service-catalog $(shell find cmd/service-catalog -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/cmd/service-catalog
 
-$(BINDIR)/user-broker: contrib/broker/k8s
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/user-broker: contrib/broker/k8s $(shell find contrib/broker/k8s -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/contrib/broker/k8s
 
 apiserver: $(BINDIR)/apiserver
-$(BINDIR)/apiserver: cmd/service-catalog
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/apiserver: cmd/service-catalog $(shell find pkg/apis/servicecatalog -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/cmd/service-catalog
 
 # Some prereq stuff
 ###################
-init: .scBuildImage .init
-
-# .init is used to know when some Glide dependencies are out of date
```

Comment:
Alright, looks like that touch pattern is a known thing.
https://www.gnu.org/software/make/manual/html_node/Empty-Targets.html